cpp/CMakeLists.txt

```
@@ -231,6 +240,12 @@ if(NOT already_set_kvikio)
       target_compile_definitions(kvikio::kvikio INTERFACE KVIKIO_CUFILE_STREAM_API_FOUND)
     endif()
   endif()
+
+  if(TARGET CURL::libcurl)
+    target_link_libraries(kvikio::kvikio INTERFACE $<BUILD_LOCAL_INTERFACE:CURL::libcurl>)
+    target_compile_definitions(kvikio::kvikio INTERFACE $<BUILD_LOCAL_INTERFACE:KVIKIO_LIBCURL_FOUND>)
```

Since this is code that gets exported out of the build, it does not need the `$<BUILD_LOCAL_INTERFACE>` genex.

```suggestion
    target_link_libraries(kvikio::kvikio INTERFACE CURL::libcurl)
    target_compile_definitions(kvikio::kvikio INTERFACE KVIKIO_LIBCURL_FOUND)
```